hr_expense_product_policy/tests/test_hr_expense.py

```diff
@@ -0,0 +1,261 @@
+                key=lambda x: x.id),
+            expected_products.sorted(key=lambda x: x.id))
+        self.department.write({
+            "allowed_expense_product_ids": [(6, 0, [self.product3.id])],
+        })
+        expected_products += self.product3
+        self.assertEqual(
+            self.employee.all_allowed_expense_product_ids.sorted(
+                key=lambda x: x.id),
+            expected_products.sorted(key=lambda x: x.id))
+
+    def test_create_expense_1(self):
+        self.employee.write({"required_expense_product": True})
+        line1 = {
+            "name": "tes line",
+            "date_value": self.expense_data["date"],
+            "unit_amount": 100.00,
+            "uom_id": self.uom1.id,
+        }
+        self.expense_data.update({
+            "required_expense_product": True,
+            "line_ids": [
+                (0, 0, line1)]})
+
+        with self.assertRaises(models.ValidationError):
+            self.obj_expense.create(self.expense_data)
+
+        self.expense_data.update({
+            "required_expense_product": False})
+
+        self.obj_expense.create(self.expense_data)
+
+    def test_create_expense_2(self):
+        self.employee.write({
+            "allowed_expense_product_ids": [(6, 0, [self.product1.id])]
+        })
+
+        line1 = {
+            "name": "tes line",
+            "date_value": self.expense_data["date"],
+            "unit_amount": 100.00,
+            "product_id": self.product2.id,
+            "uom_id": self.product2.uom_id.id,
+
+        }
+        self.expense_data.update({
+            "limit_product_selection": True,
+            "line_ids": [
+                (0, 0, line1)]})
+
+        with self.assertRaises(models.ValidationError):
+            self.obj_expense.create(self.expense_data)
+
+    def test_create_expense_3(self):
+        self.employee.write({
+            "allowed_expense_product_ids": [(6, 0, [self.product1.id])]
+        })
+
+        line1 = {
+            "name": "tes line",
+            "date_value": self.expense_data["date"],
+            "unit_amount": 100.00,
+            "product_id": self.product1.id,
+            "uom_id": self.product1.uom_id.id,
+
+        }
+        self.expense_data.update({
+            "line_ids": [
+                (0, 0, line1)]})
+
+        self.obj_expense.create(self.expense_data)
+
+    def test_write_expense_1(self):
+        self.employee.write({
+            "allowed_expense_product_ids": [(6, 0, [self.product1.id])],
+            "required_expense_product": True,
+        })
+
+        line1 = {
+            "name": "tes line",
+            "date_value": self.expense_data["date"],
+            "unit_amount": 100.00,
+            "uom_id": self.uom1.id,
+            "product_id": self.product1.id,
+        }
+        self.expense_data.update({
+            "required_expense_product": True,
+            "limit_product_selection": True,
+            "line_ids": [
+                (0, 0, line1)]})
+
```

Comment:
Don't let all of these empty lines in the middle. PEP8 encourages not to do it. Indentation is enough to make the code readable.